pkg/bytecode/vm_test.go

```diff
@@ -1041,6 +1041,149 @@ func TestErrMapKey(t *testing.T) {
+					mustMake(t, OpEqual),
+					mustMake(t, OpJumpNotTruthy, 25),
+					mustMake(t, OpConstant, 2),
+					mustMake(t, OpSetGlobal, 0),
+					mustMake(t, OpJump, 25),
+				),
+			},
+		},
+		{
+			name: "if else",
+			input: `x := 10
+			if x < 5
+				x = 20
+			else
+				x = 5
+			end`,
+			wantStackTop: makeValue(t, 5),
+			wantBytecode: &Bytecode{
+				Constants: makeValues(t, 10, 5, 20, 5),
+				Instructions: makeInstructions(
+					mustMake(t, OpConstant, 0),
+					mustMake(t, OpSetGlobal, 0),
+					mustMake(t, OpGetGlobal, 0),
+					mustMake(t, OpConstant, 1),
+					mustMake(t, OpNumLessThan),
+					mustMake(t, OpJumpNotTruthy, 25),
+					mustMake(t, OpConstant, 2),
+					mustMake(t, OpSetGlobal, 0),
+					mustMake(t, OpJump, 31),
+					mustMake(t, OpConstant, 3),
+					mustMake(t, OpSetGlobal, 0),
+				),
+			},
+		},
+		{
+			name: "else if else",
+			input: `x := 10
+			if x > 10
+				x = 10
+			else if x > 5
+				x = 5
+			else
+				x = 1
+			end`,
+			wantStackTop: makeValue(t, 5),
+			wantBytecode: &Bytecode{
+				Constants: makeValues(t, 10, 10, 10, 5, 5, 1),
+				Instructions: makeInstructions(
+					mustMake(t, OpConstant, 0),
+					mustMake(t, OpSetGlobal, 0),
+					mustMake(t, OpGetGlobal, 0),
+					mustMake(t, OpConstant, 1),
+					mustMake(t, OpNumGreaterThan),
+					mustMake(t, OpJumpNotTruthy, 25),
+					mustMake(t, OpConstant, 2),
+					mustMake(t, OpSetGlobal, 0),
+					mustMake(t, OpJump, 50),
+					mustMake(t, OpGetGlobal, 0),
+					mustMake(t, OpConstant, 3),
+					mustMake(t, OpNumGreaterThan),
+					mustMake(t, OpJumpNotTruthy, 44),
+					mustMake(t, OpConstant, 4),
+					mustMake(t, OpSetGlobal, 0),
+					mustMake(t, OpJump, 50),
+					mustMake(t, OpConstant, 5),
+					mustMake(t, OpSetGlobal, 0),
+				),
+			},
+		},
+		{
+			name: "many elseif",
+			input: `x := 3
+				if x == 1
+					x = 11
+				else if x == 2
+					x = 12
+				else if x == 3
+					x = 13
+				else 
+					x = 14
+				end`,
+			wantStackTop: makeValue(t, 13),
+			wantBytecode: &Bytecode{
+				Constants: makeValues(t, 3, 1, 11, 2, 12, 3, 13, 14),
+				Instructions: makeInstructions(
+					mustMake(t, OpConstant, 0),
+					mustMake(t, OpSetGlobal, 0),
+					mustMake(t, OpGetGlobal, 0),
+					mustMake(t, OpConstant, 1),
+					mustMake(t, OpEqual),
+					mustMake(t, OpJumpNotTruthy, 25),
+					mustMake(t, OpConstant, 2),
+					mustMake(t, OpSetGlobal, 0),
+					mustMake(t, OpJump, 69),
+					mustMake(t, OpGetGlobal, 0),
+					mustMake(t, OpConstant, 3),
+					mustMake(t, OpEqual),
+					mustMake(t, OpJumpNotTruthy, 44),
+					mustMake(t, OpConstant, 4),
+					mustMake(t, OpSetGlobal, 0),
+					mustMake(t, OpJump, 69),
+					mustMake(t, OpGetGlobal, 0),
+					mustMake(t, OpConstant, 5),
+					mustMake(t, OpEqual),
+					mustMake(t, OpJumpNotTruthy, 63),
+					mustMake(t, OpConstant, 6),
+					mustMake(t, OpSetGlobal, 0),
+					mustMake(t, OpJump, 69),
+					mustMake(t, OpConstant, 7),
+					mustMake(t, OpSetGlobal, 0),
+				),
+			},
+		},
```

Comment:
perhaps one more test where no block is executed? i.e. no else and no conditions are true.